src/compiler/checker.ts

```diff
@@ -7141,8 +7168,10 @@ namespace ts {
                     if (!target) {
                         return;
                     }
-                    let verbatimTargetName = unescapeLeadingUnderscores(target.escapedName);
-                    if (verbatimTargetName === InternalSymbolName.ExportEquals && (compilerOptions.esModuleInterop || compilerOptions.allowSyntheticDefaultImports)) {
+                    // If `target` refers to a shorthand module symbol, the name we're trying to pull out isn;t recoverable from the target symbol

```

Comment:
This is the drive-by fix, right? What’s a shorthand module symbol?

```suggestion
                    // If `target` refers to a shorthand module symbol, the name we're trying to pull out isn't recoverable from the target symbol
```

Reply:
It's a (member of a) `declare module "foo";` defined module.